unit_tests/reduct/api/entry_api_test.cc

```diff
@@ -131,6 +132,30 @@ TEST_CASE("EntryAPI::Write should write data with labels", "[api]") {
   REQUIRE(entry->BeginRead(reduct::core::Time() + us(1000001)).result->labels() == labels);
 }
 
+TEST_CASE("EntryAPI::Write should respect the passed in content-type header", "[api]") {
+  auto storage = IStorage::Build({.data_path = BuildTmpDirectory()});
+  REQUIRE(storage->CreateBucket("bucket", {}) == Error::kOk);
+
+  SECTION("write dummy record") {
+    auto [receiver, err] = EntryApi::Write(storage.get(), "bucket", "entry-1", "1000001", "18", "application/json", {});
+    REQUIRE(err == Error::kOk);
+
+    auto [resp, resp_err] = receiver(R"({"name": "random"})", true);
+    REQUIRE(resp_err == Error::kOk);
+
+    auto entry = storage->GetBucket("bucket").result.lock()->GetOrCreateEntry("entry-1").result.lock();
+  }
+
+  SECTION("check headers after reading") {
+    auto [receiver, err] = EntryApi::Read(storage.get(), "bucket", "entry-1", "1000001", {});
```

Comment:
it doesn't work because Catch's sections are independent. You have two runs here, one writes a record, and another one reads it. This code was called twice for each section and each one has its own unique data directory:

```cpp
  auto storage = IStorage::Build({.data_path = BuildTmpDirectory()});
  REQUIRE(storage->CreateBucket("bucket", {}) == Error::kOk);
```

Just remove the sections, you don't need them here.